plugins/profileChanger/profileChanger.pl

```diff
@@ -0,0 +1,133 @@
+
+sub on_unload {
+	Commands::unregister($chooks);
+}
+
+sub commandHandler {
+	my (undef, $new_profile) = @_;
+	
+	if (!$profiles::profile) {
+		error "[PC] Profiles plugin not loaded\n";
+		return;
+	}
+	
+	opendir my $d, $profiles::profile_folder;
+	my @conlist = readdir($d);
+	closedir $d;
+	
+	if (!$new_profile) {
+
+		my @profiles;
+
+		foreach (@conlist) {
+			next unless -d File::Spec->catdir($profiles::profile_folder, $_);
+			next if ($_ =~ /^\./);
+			push @profiles, $_;
+		}
+
+		@profiles = sort { $a cmp $b } @profiles;
+
+		if (@profiles) {
+			my $choice = $interface->showMenu(	#
+				"Please choose a Profiles folder.",
+				\@profiles,
+				title => "Profiles Selector"
+			);
+
+			return $quit = 1 if $choice == -1;
+
+			$new_profile = $profiles[$choice];
+		} else {
+			error "[PC] There are no profiles in profiles folder\n";
+			return;
+		}
+	} else {
+		my $found = 0;
+		foreach (@conlist) {
+			next unless -d File::Spec->catdir($profiles::profile_folder, $_);
+			next if ($_ =~ /^\./);
+			$found = 1 if ($new_profile eq $_);
+		}
+		if (!$found) {
+			error "[PC] Provided profile not found in profiles folder\n";
+			return;
+		}
+	}
+	
+	my $new_profile_folder = File::Spec->catdir($profiles::profile_folder, $new_profile);
+	my %reloadFiles;
+	
+	message "[PC] Looking for loaded files in old profile '".$profiles::profile."' to unload \n", "system";
+	foreach my $file (@{$Settings::files->getItems}) {
+		next if ($file->{'type'} != 0);
+		my $filepath;
+		if ($file->{'autoSearch'} == 1) {
+			$filepath = Settings::_findFileFromFolders($file->{'name'}, \@Settings::controlFolders);
+		} else {
+			$filepath = $file->{'name'};
+		}
+		my (undef,$directories,$filename) = File::Spec->splitpath($filepath);
+		my @dirs = File::Spec->splitdir($directories);
+		
+		if ($dirs[-2] eq $profiles::profile) {
+			message "[PC] Unloading '".$filename."' from '".$profiles::profile."'\n";
+			$reloadFiles{$file->{'index'}} = $filename;
+		}
+	}
+	
+	opendir my $d, $new_profile_folder;
+	my @newProfileFiles = readdir($d);
+	closedir $d;
+
+	message "[PC] Looking for files in new profile '".$new_profile."'\n", "system";
+	foreach my $filename (@newProfileFiles) {
+		next unless -f File::Spec->catdir($new_profile_folder, $filename);
+		next if ($filename =~ /^\./);
+		foreach my $file (@{$Settings::files->getItems}) {
+			next if ($file->{'type'} != 0);
+			next if (exists $reloadFiles{$file->{'index'}});
+			my $name = $file->{'autoSearch'} == 1 ? $file->{'name'} : $file->{'internalName'};
+			if ($name eq $filename) {
+				$reloadFiles{$file->{'index'}} = $filename;
+				message "[PC] Unloading '".$filename."' from base control folder\n";
+			}
+		}
+	}
+	
+	shift @Settings::controlFolders;
+	unshift @Settings::controlFolders, $new_profile_folder;
+
+	my $progressHandler = sub {
+		my ($filename) = @_;
+		message "[PC] Loading ".$filename."...\n";
+	};
+	
+	message "[PC] Loading files\n", "system";
+	foreach my $file_index (keys %reloadFiles) {
```

Comment:
What do you think about extracting the code in `Settings::loadByRegexp` and `Settings::loadAll` to be a generic function that we could call here?

```Perl
sub loadByRegexp {
	my ($regexp, $progressHandler) = @_;
	loadFiles([grep { $_->{name} =~ /$regexp/ } @{$files->getItems}], $progressHandler);
}

##
# void Settings::loadAll([Function progressHandler])
#
# (Re)loads all registered data files. This method follows the same contract as
# Settings::loadByHandle(), so see that method for parameter descriptions
# and exceptions.
sub loadAll {
	my ($progressHandler) = @_;
	loadFiles($files->getItems, $progressHandler);
}

sub loadFiles {
	my ($files, $progressHandler) = @_;

	Plugins::callHook('preloadfiles', {files => $files});

	my $i = 1;
	foreach my $object (@$files) {
		Plugins::callHook('loadfiles', {files => $files, current => $i});
		loadByHandle($object->{index}, $progressHandler);
		$i++;
	}

	Plugins::callHook('postloadfiles', {files => $files});
}
```